### PR TITLE
Add aria attributes to Details component

### PR DESCRIPTION
**Pull request by patrickpatrickpatrick**
## What

Add aria attributes to the Details component.

## Why

As part of the work to make the organisations page on GOV.UK more accessible. There is currently the following issue with the page:

> On the Departments, agencies and public bodies page, multiple buttons are present which began with the phrase "Works with" and contain the number of agencies or public bodies the organisation worked with. Many of these buttons have the same text occurring multiple times. For example, there are 23 occurrences of Works with 1 public body each of which relate to different department, agency or public body. Therefore, it's unclear in many instances which button related to which department, agency or public body which makes it difficult for screen reader users to differentiate between them.

@hannalaakso did some research and determined that the best way to fix the issue would be add an aria-label to the summary of the details.

[Relevant Trello Card](https://trello.com/c/NepoUk7x/1533-form-elements-on-departments-page-are-not-descriptive-of-their-function-or-purpose-9-m)

### CODE REVIEW

**Review by andysellick — COMMENTED** [review]
Needs a test 😁

**Comment by andysellick** [comment]
How urgent is this? Only I've got a PR ready to add this general functionality to all components if needed https://github.com/alphagov/govuk_publishing_components/pull/3171, which would making implementing this easier. However if it's urgent it's fine to go ahead, can rework it later.

**Comment by patrickpatrickpatrick** [comment]
> How urgent is this? Only I've got a PR ready to add this general functionality to all components if needed #3171, which would making implementing this easier. However if it's urgent it's fine to go ahead, can rework it later.

It can probably wait for that to be merged first then, I don't think it's urgent.

**Comment by patrickpatrickpatrick** [comment]
Have added a test to this now @andysellick

**Review by andysellick — APPROVED** [review]
Looks good. Please add a CHANGELOG entry before merging 👍

**Pull request merged**
